src/app/components/core/control-dispatcher/pipelines/control-pipelines.component.ts

```diff
@@ -0,0 +1,132 @@
+import { orderBy } from 'lodash';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+import { AlertDialogComponent } from '../../../common/alert-dialog/alert-dialog.component';
+import { Router } from '@angular/router';
+
+import { AlertService, ControlPipelinesService, PingService, ProgressBarService, RolesService } from '../../../../services';
+
+@Component({
+  selector: 'app-control-pipelines',
+  templateUrl: './control-pipelines.component.html',
+  styleUrls: ['./control-pipelines.component.css']
+})
+export class ControlPipelinesComponent implements OnInit, OnDestroy {
+  @ViewChild(AlertDialogComponent, { static: true }) child: AlertDialogComponent;
+  pipelines = [];
+  public showSpinner = false;
+  public childData = {};
+
+  destroy$: Subject<boolean> = new Subject<boolean>();
+
+  constructor(private controlPipelinesService: ControlPipelinesService,
+    private alertService: AlertService,
+    private ngProgress: ProgressBarService,
+    private router: Router,
+    public rolesService: RolesService) {}
+
+  ngOnInit() {
+    this.showLoadingSpinner();
+    this.getControlPipelines();
+  }
+
+  public getControlPipelines(showProgressBar = true): void {
+    /** request started */
+    if (showProgressBar) {
+      this.ngProgress.start();
+    }
+    this.controlPipelinesService.getAllPipelines()
+      .pipe(takeUntil(this.destroy$))
+      .subscribe(
+        (data: any) => {
+          /** request completed */
+          this.ngProgress.done();
+          this.pipelines = orderBy(data.pipelines, ['name'], ['asc']);
+          this.hideLoadingSpinner();
+        },
+        error => {
+          /** request completed but error */
+          this.ngProgress.done();
+          this.hideLoadingSpinner();
+          if (error.status === 0) {
+            console.log('service down ', error);
+          } else {
+            this.alertService.error(error.statusText);
+          }
+        });
+  }
+
+  public toggleDropdown(contextMenu) {
+    const id = 'dropdown-' + contextMenu;
+    const activeDropDowns = Array.prototype.slice.call(document.querySelectorAll('.dropdown.is-active'));
+    if (activeDropDowns.length > 0) {
+      if (activeDropDowns[0].id !== id) {
+        activeDropDowns[0].classList.remove('is-active');
+      }
+    }
+    const dropDown = document.querySelector(`#${id}`);
+    dropDown.classList.toggle('is-active');
+  }
+
+  deletePipeline(id) {
+    /** request started */
+    this.ngProgress.start();
+    this.controlPipelinesService.deletePipeline(id).
+      subscribe(
+        (data) => {
+          /** request completed */
+          this.ngProgress.done();
+          this.alertService.success(data['message']);
+          // delete pipeline locally from pipelines array
+          const obj = this.pipelines.filter((pipeline) => pipeline.id === id);
+          const index: number = this.pipelines.indexOf(obj[0]);
```

Comment:
use **findIndex** method to get the index of the deleted item from pipeline, instead filter and indexOf?